### PR TITLE
Add optional precision for rendering values with significant digits

### PR DESCRIPTION
**Pull request by mlohbihler**

### CODE REVIEW

**Comment by nfriedly** [comment]
Hey, thanks, that's a good idea! 

I only have my phone with me right now, but I'll take a closer look once I'm back on a computer.

**Comment by mlohbihler** [comment]
> Hey, thanks, that's a good idea!
> 
> I only have my phone with me right now, but I'll take a closer look once I'm back on a computer.

Sounds good. This would also address Issue #2 i think, although i used a different terminology.

File: test/approximate-number_test.js [hunk 1]
```diff
@@ -356,4 +356,60 @@ describe('approximate-number', function() {
...
+    // input => output
+    var tests = {
+      0: '0',
+      0.1: '0.1',
+      1: '1',
+      10: '10',
+      999: '999',
+      1000: '1k',
+      1001: '1k',
+      1234: '1.23k',
```
Comment (nfriedly) [inline test/approximate-number_test.js:374]: Some of these don't seem like what I would expect - if precision is `3`, then shouldn't this be `1.234k`?

Comment (nfriedly) [reply test/approximate-number_test.js:374]: Oh, wait, I was confusing `toFixed()` with `toPrecision()`

Comment (mlohbihler) [reply test/approximate-number_test.js:374]: It's intended to be the number of significant digits, not the number of decimal places. So you should see 1.23, 12.3, 123, 1230, etc.

**Review by nfriedly — CHANGES_REQUESTED** [review]
I left a couple of comments below, but I like this overall. One thing I noticed is that if I do 999 with precision 3, it outputs 999 (which is good). But, with precision 1 or 2, it outputs 1000, when I would expect 1k, or maybe 1.0k for precision 2.

Additionally, 9999 with precision 4 outputs 9.999k, rather than just 9999. 

I think we should maybe rework this to take precision into account before choosing the denominator.

If it helps, here's a test I wrote:

```js

  describe('options.precision > 0', function() {
    // input => output at precision 1, 2, 3, etc.
    // (precision = array index + 1)
    var tests = {
      0: ['0','0','0'],
      0.1: ['0.1', '0.1', '0.1'],
      1: ['1', '1', '1'],
      1.3: ['1', '1.3', '1.3'],
      10: ['10','10','10'],
      11: ['10','11','11'],
      999: ['1k', '1k', '999'],
      1000: ['1k', '1k', '1k'], // perhaps these should be 1k, 1.0k, 1.00k (?)
      1001: ['1k', '1k', '1k', '1001'],
      1234: ['1k', '1.2k', '1.23k', '1234'], 
      9999: ['10k', '10k', '10k', '9999'],
      10000: ['10k', '10k', '10k', '10k', '10000'], // also unsure about '10000', but I think that's what someone would expect if requesting precision 5
      // todo: add some bigger numbers
    };

    Object.keys(tests).forEach(function(input) {
      var precisionResults = tests[input];
      precisionResults.forEach(function(expected, i) {
        var precision = i + 1;
        var options = {precision: precision};
        it(format('should convert %s to %s with %j', input, expected, options), function() {
          var actual = approximateNumber(input, options);
          console.log(format('should convert %s to %s with %j got %s', input, expected, options, actual));
          assert.equal(actual, expected);
        });
      });
    });
  });
```

**Comment by mlohbihler** [comment]
> I left a couple of comments below, but I like this overall. One thing I noticed is that if I do 999 with precision 3, it outputs 999 (which is good). But, with precision 1 or 2, it outputs 1000, when I would expect 1k, or maybe 1.0k for precision 2.
> 
> Additionally, 9999 with precision 4 outputs 9.999k, rather than just 9999.
> 
> I think we should maybe rework this to take precision into account before choosing the denominator.
> 

Yeah, i like that test... very nice. Re outputs should thousands become 'k's, or should they not? Your examples look opposite to me, i.e. "would expect 1k" vs "rather than just 9999".

**Comment by nfriedly** [comment]
Sorry, I might have gotten that backwards, I wrote all that in a bit of a rush on my lunch break.

**Comment by nfriedly** [comment]
I think if precision is less than four, then thousands should become k's

**Comment by mlohbihler** [comment]
> I think if precision is less than four, then thousands should become k's

Gotchu. Pushed some changes. Please check it out when you can.

**Review by nfriedly — CHANGES_REQUESTED** [review]
Cool, this is coming along. A couple of thoughts:

* using `.precision()`essentially forces us to round rather than truncate, so we should probably document that.

* I really don't like output like `'1.001k'` and `'1.234k'`. I think that when the precision requested is equal to or greater than the number of digits in the number, we should just output the entire number without adding a k/m/whatever. (Still should add the prefix and suffix and commas and whatnot, though.)
   * Alternatively, we could just declare that precision must be in the range of 1-3

**Comment by mlohbihler** [comment]
> Cool, this is coming along. A couple of thoughts:
> 
>     * using `.precision()`essentially forces us to round rather than truncate, so we should probably document that.
> 
>     * I really don't like output like `'1.001k'` and `'1.234k'`. I think that when the precision requested is equal to or greater than the number of digits in the number, we should just output the entire number without adding a k/m/whatever. (Still should add the prefix and suffix and commas and whatnot, though.)
>       
>       * Alternatively, we could just declare that precision must be in the range of 1-3

Hi Nathan,

Please have another look when you can.

File: lib/approximate-number.js [hunk 2]
```diff
@@ -90,12 +90,14 @@
 
     if (num < thousandsBreak) {
       numString = addCommas(formatDec(num, 1, opts), opts);
+    } else if (opts.precision && opts.precision > Math.log10(num)) {
```
Comment (nfriedly) [inline lib/approximate-number.js:93]: I don't think I knew JS had a `Math.log10`

**Review by nfriedly — APPROVED** [review]
Nice work! I'll publish this shortly.

**Comment by nfriedly** [comment]
This is now published to npm in v2.1.0

**Comment by mlohbihler** [comment]
Thanks Nathan!

**Pull request merged**